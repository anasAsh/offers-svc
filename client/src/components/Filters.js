import React, { Component } from 'react';

const FILTERS = [
  {name: 'destinationName'},
  {name: 'destinationCity'},
  {name: 'regionIds', type: 'number'},
  {name: 'minTripStartDate', type: 'date'},
  {name: 'maxTripStartDate', type: 'date'},
  {name: 'lengthOfStay', type: 'number'},
  {name: 'minStarRating', type: 'number'},
  {name: 'maxStarRating', type: 'number'},
  {name: 'minTotalRate', type: 'number'},
  {name: 'maxTotalRate', type: 'number'},
  {name: 'minGuestRating', type: 'number'},
  {name: 'maxGuestRating', type: 'number'},
  {name: 'hotelId', type: 'number'}
]

class Filters extends Component {

  constructor(props) {
    super(props);
    this.handleSubmit = this.handleSubmit.bind(this);
    this.handleChange = this.handleChange.bind(this);
    this.state = {};
  }

  componentDidMount() {
    const { onSubmit } = this.props;
    onSubmit && onSubmit(this.state);
  }

  handleChange(event) {
    event.persist();
    const target = event.target;
    this.setState({[target.name]: target.value});
  }

  handleSubmit(event) {
    event.preventDefault();
    const { onSubmit } = this.props;
    onSubmit && onSubmit(this.state);
  }

  render() {
    const filtersDom = FILTERS.map(filter => {
      return <div key={filter.name}>
        <label htmlFor={filter.name}>
          {filter.label || filter.name}
        </label>
        <input
          name={filter.name}
          type={filter.type || 'text'}
          onChange={this.handleChange}
        />
      </div>
    })
    return (
      <form onSubmit={this.handleSubmit}>
        {filtersDom}
        <input type="submit" value="Submit" />
      </form>
    );
  }
}

export default Filters;
