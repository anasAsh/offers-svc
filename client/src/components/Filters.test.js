import React from "react";
import { shallow } from 'enzyme';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import Filters from "./Filters";

configure({ adapter: new Adapter() });

describe('<Filters /> component', () => {

    beforeAll(() => {
  
    });
    
    it('should match the snapshot', () => {
      const container = shallow(<Filters />);
      expect(container.html()).toMatchSnapshot();
    });

    it('should call on submit if if submit is clicked', () => {
        const jestFn = jest.fn();
        const container = shallow(<Filters onSubmit={jestFn} />);
        container.update();
        container.find('[type="submit"]').simulate('click');
        expect(jestFn).toHaveBeenCalled();
    });

    it('should call have inputs data in the submit call', async () => {
        const jestFn = jest.fn();
        const container = shallow(<Filters onSubmit={jestFn} />);
        container.update();
        container.find('[name="destinationName"]')
            .props()
            .onChange({ target: { name: 'destinationName', value: 'London' }, persist: () => {} });
        
        expect(container.state()).toEqual({destinationName: 'London'});
    });

});
