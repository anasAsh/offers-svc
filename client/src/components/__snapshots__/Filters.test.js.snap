// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Filters /> component should match the snapshot 1`] = `"<form><div><label for=\\"destinationName\\">destinationName</label><input type=\\"text\\" name=\\"destinationName\\"/></div><div><label for=\\"destinationCity\\">destinationCity</label><input type=\\"text\\" name=\\"destinationCity\\"/></div><div><label for=\\"regionIds\\">regionIds</label><input type=\\"number\\" name=\\"regionIds\\"/></div><div><label for=\\"minTripStartDate\\">minTripStartDate</label><input type=\\"date\\" name=\\"minTripStartDate\\"/></div><div><label for=\\"maxTripStartDate\\">maxTripStartDate</label><input type=\\"date\\" name=\\"maxTripStartDate\\"/></div><div><label for=\\"lengthOfStay\\">lengthOfStay</label><input type=\\"number\\" name=\\"lengthOfStay\\"/></div><div><label for=\\"minStarRating\\">minStarRating</label><input type=\\"number\\" name=\\"minStarRating\\"/></div><div><label for=\\"maxStarRating\\">maxStarRating</label><input type=\\"number\\" name=\\"maxStarRating\\"/></div><div><label for=\\"minTotalRate\\">minTotalRate</label><input type=\\"number\\" name=\\"minTotalRate\\"/></div><div><label for=\\"maxTotalRate\\">maxTotalRate</label><input type=\\"number\\" name=\\"maxTotalRate\\"/></div><div><label for=\\"minGuestRating\\">minGuestRating</label><input type=\\"number\\" name=\\"minGuestRating\\"/></div><div><label for=\\"maxGuestRating\\">maxGuestRating</label><input type=\\"number\\" name=\\"maxGuestRating\\"/></div><input type=\\"submit\\" value=\\"Submit\\"/></form>"`;
